### PR TITLE
Update important-upgrade-notes.rst

### PR DESCRIPTION
**Pull request by amyblais**
Also to send to GitLab.

### CODE REVIEW

**Comment by amyblais** [comment]
Add: config changes re channel sidebar: https://github.com/mattermost/docs/pull/4266/files

**Comment by amyblais** [comment]
@agnivade Is the go update to 1.15.5 something that should also be documented here? (It's already in the changelog)

**Comment by agnivade** [comment]
IMO having it in release notes should be sufficient.

File: source/administration/important-upgrade-notes.rst [hunk 1]
```diff
@@ -8,14 +8,21 @@ Important Upgrade Notes
...
+|                                                    | functional if the Legacy Sidebar (``EnableLegacySidebar``) is enabled since they are not compatible with the new sidebar experience.                             |
+|                                                    | ``ExperimentalChannelSidebarOrganization`` has been deprecated, since the new sidebar is now enabled for all users.                                              |
+|                                                    +------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+|                                                    | Breaking changes to the Golang client API were introduced. Any client built on top of this client that makes use of functions for signature change (e.g. for     |
+|                                                    | changing the number of parameters or their type) will fail to compile/execute as soon as its dependencies get updated.                                           |
```
Comment (streamer45) [inline source/administration/important-upgrade-notes.rst:24]: I'd change this section to:

```
Breaking changes to the Golang client API were introduced:
GetPostThread, GetPostsForChannel, GetPostsSince, GetPostsAfter, GetPostsBefore, GetPostsAroundLastUnread now require an additional collapsedThreads parameter to be passed. 
Any client making use of these functions will need to update them when upgrading its dependencies.
```

**Review by streamer45 — APPROVED** [review]
Looks good :+1:

File: source/administration/important-upgrade-notes.rst [hunk 2]
```diff
@@ -8,14 +8,22 @@ Important Upgrade Notes
...
 
 +----------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
 | If you’re upgrading from a version earlier than... | Then...                                                                                                                                                          |
 +====================================================+==================================================================================================================================================================+
+| v5.32.0                                            | ``ExperimentalChannelOrganization``, ``EnableXToLeaveChannelsFromLHS``, ``CloseUnusedDirectMessages``, and ``ExperimentalHideTownSquareinLHS`` settings are only |
+|                                                    | functional if the Legacy Sidebar (``EnableLegacySidebar``) is enabled since they are not compatible with the new sidebar experience.                             |
```
Comment (esethna) [inline source/administration/important-upgrade-notes.rst:20]: @amyblais please add a link embeded somewhere in here to the blog that will be published on the new sidebar: https://mattermost.com/blog/custom-collapsible-channel-categories/

**Review by esethna — CHANGES_REQUESTED** [review]

**Review by wiersgallak — APPROVED** [review]
LGTM

**Pull request merged**